### ISSUE TITLE
Why concatenate slots in a table with <SEP> & <EOS>?

### ISSUE THREAD

**Issue by insomnia1996** [yxuansu/Few-Shot-Table-to-Text-Generation#2]
Thanks for releasing your code! After reading your paper, however, I was a little confused about the preprocessing method for the input tables.

> prototype_selector/dataclass.py line 74:
```
one_res_str = slot_key + ' ' + SEP + ' ' + slot_value + ' ' + EOS
```
Why do you use SEP and EOS to concatenate every key and value in the table? In my opinion,  you should feed no more than 2 sentences with only one EOS into BERT. 

I also wonder that you randomly retrieve 100 sentences using IR system to build _**R**_ . Does that mean you are able to access all the gold summaries in the dev/test set, for we only have 100 samples in the train set?

Thanks in advance.

**Comment by yxuansu** [comment]
> Thanks for releasing your code! After reading your paper, however, I was a little confused about the preprocessing method for the input tables.
> 
> > prototype_selector/dataclass.py line 74:
> 
> ```
> one_res_str = slot_key + ' ' + SEP + ' ' + slot_value + ' ' + EOS
> ```
> 
> Why do you use SEP and EOS to concatenate every key and value in the table? In my opinion, you should feed no more than 2 sentences with only one EOS into BERT.
> 
> I also wonder that you randomly retrieve 100 sentences using IR system to build _**R**_ . Does that mean you are able to access all the gold summaries in the dev/test set, for we only have 100 samples in the train set?
> 
> Thanks in advance.

Thank you for your questions! Sorry for my late reply... The reason why we use this input format is to let the model know exactly which part is slot key and which part is slot value. As we are using a BART or T5 model to produce the result, we can just like to model learn to identify the input based on the specific input format. 

For your second question, during the retrieval stage, we manually filtering out the reference summary from the retrieved results. So the candidates R does not contain the reference summary to avoid the data-leakage problem.

**Comment by insomnia1996** [comment]
> > Thanks for releasing your code! After reading your paper, however, I was a little confused about the preprocessing method for the input tables.
> > > prototype_selector/dataclass.py line 74:
> > 
> > 
> > ```
> > one_res_str = slot_key + ' ' + SEP + ' ' + slot_value + ' ' + EOS
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Why do you use SEP and EOS to concatenate every key and value in the table? In my opinion, you should feed no more than 2 sentences with only one EOS into BERT.
> > I also wonder that you randomly retrieve 100 sentences using IR system to build _**R**_ . Does that mean you are able to access all the gold summaries in the dev/test set, for we only have 100 samples in the train set?
> > Thanks in advance.
> 
> Thank you for your questions! Sorry for my late reply... The reason why we use this input format is to let the model know exactly which part is slot key and which part is slot value. As we are using a BART or T5 model to produce the result, we can just like to model learn to identify the input based on the specific input format.
> 
> For your second question, during the retrieval stage, we manually filtering out the reference summary from the retrieved results. So the candidates R does not contain the reference summary to avoid the data-leakage problem.

Thanks a lot for your reply! Is data/human/train_summary_top_100.txt retrieved from the 500 training instances? What hyper-parameter _top_n_ do you use when you only have 20, 50 or 100 training samples?

**Comment by yxuansu** [comment]
> > > Thanks for releasing your code! After reading your paper, however, I was a little confused about the preprocessing method for the input tables.
> > > > prototype_selector/dataclass.py line 74:
> > > 
> > > 
> > > ```
> > > one_res_str = slot_key + ' ' + SEP + ' ' + slot_value + ' ' + EOS
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Why do you use SEP and EOS to concatenate every key and value in the table? In my opinion, you should feed no more than 2 sentences with only one EOS into BERT.
> > > I also wonder that you randomly retrieve 100 sentences using IR system to build _**R**_ . Does that mean you are able to access all the gold summaries in the dev/test set, for we only have 100 samples in the train set?
> > > Thanks in advance.
> > 
> > 
> > Thank you for your questions! Sorry for my late reply... The reason why we use this input format is to let the model know exactly which part is slot key and which part is slot value. As we are using a BART or T5 model to produce the result, we can just like to model learn to identify the input based on the specific input format.
> > For your second question, during the retrieval stage, we manually filtering out the reference summary from the retrieved results. So the candidates R does not contain the reference summary to avoid the data-leakage problem.
> 
> Thanks a lot for your reply! Is data/human/train_summary_top_100.txt retrieved from the 500 training instances? What hyper-parameter _top_n_ do you use when you only have 20, 50 or 100 training samples?

Hey,  train_summary_top_100.txt contains retrieved results for 500 tables. For each table, 100 summaries are retrieved from the whole wiki corpus. The corresponding training table can be found in the train_table.txt file.

**Comment by insomnia1996** [comment]
> > > > Thanks for releasing your code! After reading your paper, however, I was a little confused about the preprocessing method for the input tables.
> > > > > prototype_selector/dataclass.py line 74:
> > > > 
> > > > 
> > > > ```
> > > > one_res_str = slot_key + ' ' + SEP + ' ' + slot_value + ' ' + EOS
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > Why do you use SEP and EOS to concatenate every key and value in the table? In my opinion, you should feed no more than 2 sentences with only one EOS into BERT.
> > > > I also wonder that you randomly retrieve 100 sentences using IR system to build _**R**_ . Does that mean you are able to access all the gold summaries in the dev/test set, for we only have 100 samples in the train set?
> > > > Thanks in advance.
> > > 
> > > 
> > > Thank you for your questions! Sorry for my late reply... The reason why we use this input format is to let the model know exactly which part is slot key and which part is slot value. As we are using a BART or T5 model to produce the result, we can just like to model learn to identify the input based on the specific input format.
> > > For your second question, during the retrieval stage, we manually filtering out the reference summary from the retrieved results. So the candidates R does not contain the reference summary to avoid the data-leakage problem.
> > 
> > 
> > Thanks a lot for your reply! Is data/human/train_summary_top_100.txt retrieved from the 500 training instances? What hyper-parameter _top_n_ do you use when you only have 20, 50 or 100 training samples?
> 
> Hey, train_summary_top_100.txt contains retrieved results for 500 tables. For each table, 100 summaries are retrieved from the whole wiki corpus. The corresponding training table can be found in the train_table.txt file.

I got it. Thank you for your patience.
